tests/console/yast2_samba.pm

```diff
@@ -0,0 +1,279 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use strict;
+use base "consoletest";
+use testapi;
+
+
+
+sub run() {
+    select_console 'root-console';
+
+    # check network at first
+    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
+    # install local ldap server
+    assert_script_run("/usr/bin/zypper -n -q in yast2-auth-server yast2-ldap openldap2 openldap2-client krb5-server krb5-client tdb-tools");
+
+    script_run("/sbin/yast2 auth-server; echo yast2-auth-server-status-\$? > /dev/$serialdev", 0);
+    # check ldap server configuration started
+    assert_screen 'yast2_ldap_configuration_startup';
+    send_key 'alt-f';
+    assert_screen 'yast2_ldap_configuration_general-setting_firewall';
+    sleep 1;
+    send_key 'alt-e';
+
+    # configure stand-alone ldap server
+    assert_screen 'yast2_ldap_configuration_stand-alone';
+    send_key 'alt-n';
+    sleep 1;
+    assert_screen 'yast2_ldap_configuration_stand-alone_tls';
+
+    # move to next page basic database settings and set base dn, ldap admin password
+    send_key 'alt-n';
+    sleep 1;
+    send_key 'alt-s';
+    sleep 1;
+    type_string 'dc=qa, dc=suse, dc=de';
+    sleep 1;
+    send_key 'alt-a';
+    sleep 1;
+    for (1 .. 20) { send_key 'backspace'; }
+    type_string 'cn=admin';
+    send_key 'alt-l';
+    type_string 'testing';
+    sleep 1;
+    send_key 'alt-v';
+    type_string 'testing';
+    sleep 1;
+    # use database as default for ldap client
+    send_key 'alt-u';
+    sleep 1;
+    send_key 'alt-n';
+    sleep 1;
+    assert_screen 'yast2_ldap_configuration_kerberos';
+    send_key 'alt-x';
+    sleep 1;
+    assert_screen 'yast2_ldap_configuration_summary';
+
+    # finish ldap server configuration
+    send_key 'alt-f';
+    sleep 10;
+
+    # check ldap server status at first, a local ldap server is needed in the test case
+    assert_script_run "systemctl show -p ActiveState slapd.service | grep ActiveState=active";
+
+    # install samba stuffs at first
+    assert_script_run("/usr/bin/zypper -n -q in samba samba-client yast2-samba-server");
```

Comment:
the manual doesn't talk about installing samba or samba-client..would it be better to have the test assume the same as the manual does?

https://www.suse.com/documentation/sles11/book_sle_admin/data/sec_samba_serv_inst.html#sec_samba_yast2_conf_inst